sqlutil/monitor.go

```diff
@@ -0,0 +1,56 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+)
+
+// MonitorDatabase collects database connection pool metrics at regular intervals
+func MonitorDatabase(
+	ctx context.Context,
+	conf *config.Config,
+	statsFactory stats.Stats,
+	db *sql.DB,
+	identifier string,
+) {
+	reportInterval := conf.GetDurationVar(10, time.Second, "Database.ReportInterval")
```

Comment:
We can make this name a bit more specific, for instance: Database stats report interval.